sycl/plugins/cuda/pi_cuda.cpp

```diff
@@ -3481,8 +3519,12 @@ pi_result cuda_piextUSMEnqueueMemset(pi_queue queue, void *ptr, pi_int32 value,
                                      pi_uint32 num_events_in_waitlist,
                                      const pi_event *events_waitlist,
                                      pi_event *event) {
+  // enforce that the queue is valid
   assert(queue != nullptr);
-  assert(ptr != nullptr);
+  // check that the pointer is valid
+  if (ptr == nullptr) {
```

Comment:
Please, return assert.

Reply:
No - the whole point is that the code calling this expects an error, not an `abort()`.
Using an `assert()` here and below will FAIL the USM tests.

Reply:
This is necessary to pass the following USM test: https://github.com/intel/llvm/blob/sycl/sycl/test/usm/memset.cpp#L49

Reply:
I guess that the SYCL runtime implementation could be changed to check arguments before calling into the plugins?

Reply:
Ok, agree. SYCL runtime implementation can be changed, but it is a subject for a separate PR.